iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -159,9 +176,14 @@ struct StopDetailsPage: View {
             } else {
                 predictionsRepository.connect(stopIds: [stop.id]) { outcome in
                     DispatchQueue.main.async {
+                        let errorKey = "StopDetailsPage.joinPredictions"
                         switch onEnum(of: outcome) {
-                        case let .ok(result): predictions = result.data
-                        case .error: predictions = nil
+                        case let .ok(result):
+                            errorBannerRepository.clearDataError(key: errorKey)
+                            predictions = result.data
+                        case .error:
+                            errorBannerRepository.setDataError(key: errorKey) { loadEverything() }
+                            predictions = nil
```

Comment:
issue: Testing out locally on stop details with the following steps, data doesn't reload as I'd expect it to.
1. Start on nearby transit
2. Turn on airplane mode
3. Navigate into stop
4. See error message as expected
5. Turn off airplane mode
6. Click refresh button
7. Data doesn't load
8. Go back to nearby transit, then back into the same stop, data loads as expected